clang/lib/Sema/SemaTemplateInstantiateDecl.cpp

```diff
@@ -5271,7 +5271,8 @@ void Sema::InstantiateVariableDefinition(SourceLocation PointOfInstantiation,
       : Consumer(Consumer), Var(Var) { }
 
     ~PassToConsumerRAII() {
-      Consumer.HandleCXXStaticMemberVarInstantiation(Var);
+      if (Var)
```

Comment:
Is this change necessary? Does HandleCXXStaticMemberVarinstantiation not peroperly handle the null case?

Just surprised that none of the rest of this got there.

Reply:
Also, It seems that it would be more readable to put the logic from below here, isntead of 'if (var)', just 'f(!SyclIsDevice || !Var->isStaticDataMember())

Reply:
> Is this change necessary? Does HandleCXXStaticMemberVarinstantiation not peroperly handle the null case?

No it straight up dereferences the incoming VD.  I could add a NULL check there if you want me to.

Reply:
> Also, It seems that it would be more readable to put the logic from below here, isntead of 'if (var)', just 'f(!SyclIsDevice || !Var->isStaticDataMember())

I couldn't how to access getLangOpts() in the local struct without passing the this pointer from the enclosing member function.

Reply:
*figure out

Reply:
Ah, right, I missread ASTConsumer as ASTcontext.  Just give it a reference to Sema?

Reply:
Done.  Thanks Erich!